### PR TITLE
Round EEGLAB latencies

### LINKED ISSUE #4954: dealing with eeglab float event latencies
Eeglab represents event latencies with floats after downsampling (to suggest that events are no longer precisely co-occuring with specific time samples). Instead of turning these to int representation, which truncates the value, we should be rounding.

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Closes #4954

@mmagnuski

### CODE REVIEW

**Comment by larsoner** [comment]
We also need to only `warn` if we get one that is `== -1`, and raise error if any are `< -1`

**Comment by jona-sassenhagen** [comment]
I'm not sure I completely understood the issue. What if the sample index is .7?

**Comment by larsoner** [comment]
0.7 will round to 1.0, get -1'ed to 0 and be fine

0.4 will round to 0., get -1'ed to -1 and emit a warning

So yes the warning vs not will depend on the level of downsampling but really we just want to make sure people inspect their data in the case that there is a -1 (we are not trying to warn about fractional samples themselves) so it's okay

**Comment by jona-sassenhagen** [comment]
Uh sorry, I meant, what if it's -.7. It'll get rounded to -1, and then ..?

**Comment by larsoner** [comment]
then it gets -1'ed to -2 and there is an error. `EEGLAB` should not give such negative samples

**Comment by jona-sassenhagen** [comment]
Ahhhh right.

Did I ever mention I almost didn't get my high school diploma because I got a really bad math grade?

**Comment by jona-sassenhagen** [comment]
Should I add a test? I guess it would be one of these where we need to create and write a mat file, i.e., longer than the rest of this PR ...

File: mne/io/eeglab/eeglab.py [hunk 1]
```diff
@@ -667,9 +667,12 @@ def read_events_eeglab(eeg, event_id=None, event_id_func='strip_to_integer',
         logger.info('No events found, returning empty stim channel ...')
         return np.zeros((0, 3))
 
-    if (latencies < 0).any():
+    if (latencies < -1).any():
         raise ValueError('At least one event sample index is negative. Please'
                          ' check if EEG.event.sample values are correct.')
+    if latencies == -1).any():
```
Comment (larsoner) [inline mne/io/eeglab/eeglab.py]: syntax error

**Comment by larsoner** [comment]
Yes a test would be good and probably not so terrible to add actually. Wait until my EEGLAB PR is merged and it should be a one-line change to one of the existing `savemat` calls followed by a `with pytest.warns(RuntimeWarning, match="negative):` or so

**Comment by jona-sassenhagen** [comment]
Ok

**Comment by jona-sassenhagen** [comment]
@larsoner , you meant this one right: #5326

So I should finish this one now?

**Comment by jona-sassenhagen** [comment]
ugh what did I do

**Pull request closed without merging**